packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -84,71 +126,48 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
      */
     function getFundingRateForContract(address perpetual)
         external
-        view
         override
-        nonReentrantView()
+        nonReentrant()
+        publishAndWithdrawProposal(perpetual)
         returns (FixedPoint.Signed memory)
     {
         FundingRateRecord storage fundingRateRecord = _getFundingRateRecord(perpetual);
-
-        if (_getProposalState(fundingRateRecord.proposal) == ProposalState.Expired) {
-            return fundingRateRecord.proposal.rate;
-        } else {
-            return fundingRateRecord.rate;
-        }
+        return fundingRateRecord.rate;
     }
 
     /**
      * @notice Returns the timestamp at which the current funding rate was proposed.
-     * @dev The "current funding rate" is defined as that returned by `getFundingRateForIdentifier(id)`
      * @param perpetual Perpetual to retrieve propose time for.
      * @return propose timestamp.
      */
-    function getProposeTimeForContract(address perpetual) external view nonReentrantView() returns (uint256) {
-        return _getLatestProposeTimeForContract(perpetual);
+    function getProposeTimeForContract(address perpetual)
```

Comment:
This method is only used for testing, am open to removing it. The main feature it provides is that the caller can get an up to date proposeTime after any pending proposals are published

Reply:
I am going to remove it, as its not part of any interface

Reply:
Yeah, agree with @chrismaree that if you just need it for testing, then we can create a test contract to expose the info. Prefer to remove anything that we don't see a use for in a production context.

Reply:
I removed .